Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# use fout.write(f"...") to save the data 
def exp_expansion(x, N, fname):
    """Add docs here"""
    total = 0.0
    fout = open(fname, "w")
    # YOUR CODE HERE
    raise NotImplementedError()
    fout.close()

def factorial(n):
    """Add docs here"""
    result = 1.0
    for ii in range(1,n+1):
        result *= ii
    return result
 
# Call the function  
exp_expansion(x=1.8766, N=10000, fname="sumdata.txt")

------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mNotImplementedError[0m                       Traceback (most recent call last)
Cell [0;32mIn[3], line 18[0m
[1;32m     15[0m     [38;5;28;01mreturn[39;00m result
[1;32m     17[0m [38;5;66;03m# Call the function  [39;00m
[0;32m---> 18[0m [43mexp_expansion[49m[43m([49m[43mx[49m[38;5;241;43m=[39;49m[38;5;241;43m1.8766[39;49m[43m,[49m[43m [49m[43mN[49m[38;5;241;43m=[39;49m[38;5;241;43m10000[39;49m[43m,[49m[43m [49m[43mfname[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;124;43msumdata.txt[39;49m[38;5;124;43m"[39;49m[43m)[49m

Cell [0;32mIn[3], line 7[0m, in [0;36mexp_expansion[0;34m(x, N, fname)[0m
[1;32m      5[0m fout [38;5;241m=[39m [38;5;28mopen[39m(fname, [38;5;124m"[39m[38;5;124mw[39m[38;5;124m"[39m)
[1;32m      6[0m [38;5;66;03m# YOUR CODE HERE[39;00m
[0;32m----> 7[0m [38;5;28;01mraise[39;00m [38;5;167;01mNotImplementedError[39;00m()
[1;32m      8[0m fout[38;5;241m.[39mclose()

[0;31mNotImplementedError[0m: 

